test/jb_token/mint.test.js

```diff
@@ -0,0 +1,60 @@
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+
+describe('JBToken::mint(...)', function () {
+  const PROJECT_ID = 10;
+  const name = 'TestTokenDAO';
+  const symbol = 'TEST';
+
+  let deployer;
+  let addrs;
+  let testToken;
+
+  before(async function () {
+    [deployer, ...addrs] = await ethers.getSigners();
+    testToken = await deployToken(name, symbol);
+  });
+
+  async function deployToken(name, symbol) {
+    const jbTokenFactory = await ethers.getContractFactory('JBToken');
+    const jbToken = await jbTokenFactory.deploy(name, symbol);
+    await jbToken.deployed();
+    return jbToken;
+  }
+
+  // Tests
+
+  it('Should mint token and emit event if caller is owner', async function () {
+    const addr = addrs[1];
+    const numTokens = 3000;
+    const mintTx = await testToken.connect(deployer).mint(PROJECT_ID, addr.address, numTokens);
+
+    await expect(mintTx).to.emit(testToken, 'Transfer');
+
+    // overloaded functions need to be called using the full function signature
+    const balance = await testToken['balanceOf(uint256,address)'](PROJECT_ID, addr.address);
```

Comment:
Let's call the functions directly.

await testToken.balanceOf(PROJECT_ID, addr.address)

Reply:
I tried that first but since this is an overloaded function hardhat can't recognize that way of calling balanceOf